src/App.tsx

```diff
@@ -9,39 +9,31 @@
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
-import { getUserPosts } from './api/posts';
-import { User } from './types/User';
-import { Post } from './types/Post';
+import { useAppDispatch, useAppSelector } from './app/hooks';
+import * as usersActions from './features/users';
+import { actions as selectedPostActions } from './features/selectedPost';
 
 export const App: React.FC = () => {
-  const [posts, setPosts] = useState<Post[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
+  const dispatch = useAppDispatch();
 
-  const [author, setAuthor] = useState<User | null>(null);
-  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const { hasError } = useAppSelector(state => state.users);
+  const { post: selectedPost } = useAppSelector(state => state.selectedPost);
+  const {
+    author,
+    loaded: postLoaded,
+    posts,
+    hasError: postError,
+  } = useAppSelector(state => state.author);
 
-  function loadUserPosts(userId: number) {
-    setLoaded(false);
-
-    getUserPosts(userId)
-      .then(setPosts)
-      .catch(() => setError(true))
-      // We disable the spinner in any case
-      .finally(() => setLoaded(true));
-  }
+  useEffect(() => {
+    dispatch(usersActions.init());
+  }, [dispatch]);
 
   useEffect(() => {
-    // we clear the post when an author is changed
-    // not to confuse the user
-    setSelectedPost(null);
+    dispatch(selectedPostActions.clearPost());
+  }, [author, dispatch]);
 
-    if (author) {
-      loadUserPosts(author.id);
-    } else {
-      setPosts([]);
-    }
-  }, [author]);
+  const postErrorCheck = postLoaded && !hasError && !postError && author;
```

Comment:
The `postErrorCheck` logic is a bit complex. Consider breaking it down or adding comments to clarify the conditions being checked. This will improve readability and maintainability.